file.go

```diff
@@ -293,15 +293,25 @@ func (f *File) reload(s dataSource) error {
 
 // Reload reloads and parses all data sources.
 func (f *File) Reload() (err error) {
-	for _, s := range f.dataSources {
+	for i, s := range f.dataSources {
 		if err = f.reload(s); err != nil {
+			// Notice: if ShortCircuit and Loose are both true, ShortCircuit has priority over Loose.
+			// Return err when all configuration sources fail to load.
+			if f.options.ShortCircuit {
+				if i < len(f.dataSources)-1 {
+					continue
+				}
+			}
 			// In loose mode, we create an empty default section for nonexistent files.
 			if os.IsNotExist(err) && f.options.Loose {
 				_ = f.parse(bytes.NewBuffer(nil))
 				continue
 			}
 			return err
 		}
+		if f.options.ShortCircuit {
+			return nil
+		}
```

Comment:
I think these lines no longer useful?

Reply:
I think it is useful. When the two loaded configuration sources are valid, a null message is returned immediately after the first valid configuration source is loaded, and the loading of the second configuration source is stopped.
If this code is removed, both configuration sources are loaded.

Reply:
Ah, right right.